### PR TITLE
Rule and Module Configurations are now copied + some fixes in Provide…

### PR DESCRIPTION
**Pull request by adimova**
…rs(#2976, #2977)

Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.integration.test/src/test/groovy/org/eclipse/smarthome/automation/integration/test/AutomationIntegrationJsonTest.groovy [hunk 1]
```diff
@@ -249,7 +250,8 @@ class AutomationIntegrationJsonTest extends OSGiTest{
         def trigger = rule.triggers.find{it.id.equals("ItemStateChangeTriggerID")} as Trigger
         assertThat trigger, is(notNullValue())
         assertThat trigger.typeUID, is("core.GenericEventTrigger")
-        assertThat trigger.configuration.get("eventSource"), is ("myMotionItem")
+        //assertThat trigger.configuration.get("eventSource"), is ("myMotionItem") -> after bugfix
+        // the only way to check this is to run the rule and check that it is executed successfully
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.integration.test/src/test/groovy/org/eclipse/smarthome/automation/integration/test/AutomationIntegrationJsonTest.groovy]: I don't think we need the comment out line at all (if you add a note in the long description of the commit it should be enough if someone read the history to find the reason why the line has been removed).

File: bundles/automation/org.eclipse.smarthome.automation.integration.test/src/test/groovy/org/eclipse/smarthome/automation/integration/test/AutomationIntegrationJsonTest.groovy [hunk 2]
```diff
@@ -260,10 +262,44 @@ class AutomationIntegrationJsonTest extends OSGiTest{
         def action = rule.actions.find{it.id.equals("ItemPostCommandActionID")} as Action
         assertThat action, is(notNullValue())
         assertThat action.typeUID, is("core.ItemCommandAction")
-        assertThat action.configuration.get("itemName"), is("myLampItem")
+        //assertThat action.configuration.get("itemName"), is("myLampItem") -> after bugfix
+        // the only way to check this is to run the rule and check that it is executed successfully
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.integration.test/src/test/groovy/org/eclipse/smarthome/automation/integration/test/AutomationIntegrationJsonTest.groovy]: I don't think we need the comment out line at all (if you add a note in the long description of the commit it should be enough if someone read the history to find the reason why the line has been removed).

**Comment by danchom** [comment]
LGTM

**Comment by maggu2810** [comment]
> LGTM

@danchom You could start using the Github review(er) feature to give your approval. It will be added to the reviewer section with your "ack" / "no ack" status.

**Comment by danchom** [comment]
OK thanks @maggu2810

**Pull request merged**